airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/DestinationConsumerCallback.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.base;
+
+import io.airbyte.commons.lang.CloseableQueue;
+
+public interface DestinationConsumerCallback {
+
+  void queryDatabase(String query) throws Exception;
+
+  void queryDatabaseInTransaction(String queries) throws Exception;
+
+  void writeBufferedRecords(int batchSize, CloseableQueue<byte[]> writeBuffer, String schemaName, String tmpTableName);
+
+  String createRawTableQuery(String schemaName, String streamName);
```

Comment:
can this just be `createTableQuery`? it takes in a schema name and a stream name and it creates that table. doesn't need to know if it's going to be a raw one or not.

Reply:
Sure, but it's always going to create a table with the three columns (ab_id, data and emitted_at), not just any generic tables.

Reply:
good point. i think "raw" is the part that seems wrong. what you changed it to seems good to me. if you prefer calling it an airbyte table or something that's fine too.